lib/pacemaker/pcmk_scheduler.c

```diff
@@ -730,45 +730,15 @@ log_unrunnable_actions(const pcmk_scheduler_t *scheduler)
     }
 }
 
-/*!
- * \internal
- * \brief Unpack the CIB for scheduling
- *
- * \param[in,out] cib        CIB XML to unpack (may be NULL if already unpacked)
- * \param[in]     flags      Scheduler flags to set in addition to defaults
- * \param[in,out] scheduler  Scheduler data
- */
-static void
-unpack_cib(xmlNode *cib, unsigned long long flags, pcmk_scheduler_t *scheduler)
-{
-    if (pcmk_is_set(scheduler->flags, pcmk__sched_have_status)) {
-        crm_trace("Reusing previously calculated cluster status");
-        pcmk__set_scheduler_flags(scheduler, flags);
-        return;
-    }
-    pcmk__assert(cib != NULL);
-    crm_trace("Calculating cluster status");
-    pcmk_reset_scheduler(scheduler);
-    pcmk__set_scheduler_flags(scheduler, flags);
-    scheduler->input = cib;
-    cluster_status(scheduler); // Sets pcmk__sched_have_status
-}
-
 /*!
  * \internal
  * \brief Run the scheduler for a given CIB
  *
- * \param[in,out] cib        CIB XML to use as scheduler input
- * \param[in]     flags      Scheduler flags to set in addition to defaults
  * \param[in,out] scheduler  Scheduler data
  */
 void
-pcmk__schedule_actions(xmlNode *cib, unsigned long long flags,
```

Comment:
We could keep the `flags` argument for convenience and save a handful of lines in the callers. However, to me it seems clearer to set the flags in the callers (and sometimes it gives better trace output from the `pcmk__set_scheduler_flags()` macro).

The later commit that re-adds a `cluster_status()` call at the beginning of `pcmk__schedule_actions()` is different, because `pcmk__schedule_actions()` actually expects or requires the status to be set before scheduling actions. There is no such need for particular flags to be set.